src/variables.jl

```diff
@@ -34,6 +34,39 @@ done in the current state of the model `model`.
 """
 add_variable(model::ModelLike) = throw(AddVariableNotAllowed())
 
+"""
+    supports_add_constrained_variable(
+        model::ModelLike,
+        S::Type{<:AbstractScalarSet}
+    )::Bool
+
+Return a `Bool` indicating whether `model` supports constraining a variable
+to belong to a set of type `S` either on creation of the variable with
+[`add_constrained_variable`](@ref) or after the variable is created with
+[`add_constraint`](@ref).
+
+By default, this function falls back to
+`supports_add_constrained_variables(model, Reals) &&
+supports_constraint(model, MOI.SingleVariable, S)` which is the correct
+definition for most models.
+
+## Example
+
+Suppose that a solver supports only two kind of variables: binary variables
+and continuous variables with a lower bound. If the solver decide not to
+support `SingleVariable`-in-`Binary` and `SingleVariable`-in-`GreaterThan`
+constraints, it only has to implement `add_constrained_variable` for these
+two sets which prevents the user to add both a binary constraint and a
+lower bound on the same variable. Moreover, if the user adds a
+`SingleVariable`-in-`GreaterThan` constraint, it will transparently be bridged
```

Comment:
"it will be transparently bridged" depends on how bridges are set up. It's more precise to say that implementing this interface (i.e., `supports_add_constrained_variables`) enables the constraint to be transparently bridged.